app/xds_api/serializers.py

```diff
@@ -124,3 +125,48 @@ def validate(self, data):
 
         # returns when user is inactive or not in the system
         raise serializers.ValidationError('Incorrect Credentials')
+
+
+class CourseSerializer(serializers.ModelSerializer):
+    """Serializes the Course model"""
+    class Meta:
+        model = Course
+        fields = ['metadata_key_hash']
+
+
+class InterestListSerializer(serializers.ModelSerializer):
+    """Serializes the interest list model"""
+    owner = XDSUserSerializer(read_only=True)
+    subscribers = XDSUserSerializer(many=True, read_only=True)
+
+    class Meta:
+        model = InterestList
+        fields = '__all__'
+
+    def create(self, validated_data):
+        name = validated_data.get("name")
+        description = validated_data.get("description")
+        owner = validated_data.get("owner")
+        return InterestList.objects.create(name=name,
+                                           description=description,
+                                           owner=owner)
+
+    def update(self, instance, validated_data):
+        instance.owner = validated_data.get('owner', instance.owner)
+        instance.description = validated_data.get('description',
+                                                  instance.description)
+        instance.name = validated_data.get('name', instance.name)
+        courses = validated_data.get('courses')
+
+        # add all new course
```

Comment:
Nit: Sort of a redundant comment -- if we think it's useful to have a comment here we should probably add some more context.  Otherwise I'd just remove it and describe what is generally happening in a function docstring.